Homework/homework 2

VictorKozhevnikov opened the pull request.

VictorKozhevnikov committed:
Merge branch 'homework/homework-1' into develop

VictorKozhevnikov committed:
introduce tslint

VictorKozhevnikov committed:
add linting to build setup and fix mistakes

VictorKozhevnikov committed:
work in progress

VictorKozhevnikov committed:
homework 2

ubcent commented on src/dir-watcher/dir-watcher-instance.ts:
```diff
@@ -0,0 +1,24 @@
+        private readonly fsWatcher: FSWatcher,
+        private readonly basePath: string
+    ) {
+        this.eventEmitter = new EventEmitter();
```
You should extend EventEmmiter instead of using it

ubcent commented on src/dir-watcher/dir-watcher-instance.ts:
```diff
@@ -0,0 +1,24 @@
+import { FSWatcher } from 'fs';
+import * as path from 'path';
+
+export class DirWatcherInstance {
```
Why do you need additional class? It looks like interface or abstract class

VictorKozhevnikov committed:
Updates after code review
 - Inherit DirWatcher from EventEmitter
 - Inherit Importer from DirWatcher
 - pass watch parameters to DirWatchr's constructor

VictorKozhevnikov committed:
Introduced interfaces IDirWatcher and IImporter

VictorNovice commented:
Hi, I've pushed the updates, see now
